src/test/scala/is/hail/utils/RichRDDSuite.scala

```diff
@@ -28,4 +28,28 @@ class RichRDDSuite extends SparkSuite {
     val vds = hc.importVCF("src/test/resources/sample.vcf")
     assert(vds.head(3).countVariants() == 3)
   }
+
+  @Test def binaryParallelWrite() {
+    def readBytes(file: String): Array[Byte] = hadoopConf.readFile(file) { dis =>
+      val buffer = new Array[Byte](32)
+      val size = dis.read(buffer)
+      buffer.take(size)
+    }
+
+    val header = Array[Byte](108, 27, 1, 91)
+    val data = Array(Array[Byte](1, 19, 23, 127, -1), Array[Byte](23, 4, 15, -2, 1))
+    val r = sc.parallelize(data, numSlices = 2)
+    assert(r.getNumPartitions == 2)
+
+    val notParallelWrite = tmpDir.createTempFile("notParallelWrite")
+    r.saveFromByteArrays(notParallelWrite, tmpDir.tempDir, Some(header), parallelWrite = false)
+
+    assert(readBytes(notParallelWrite) sameElements header ++: data.flatten)
```

Comment:
Since the types are the same, is `++:` any different from `++`? I think I prefer `++` because it's less unusual and people won't have to look up how `++:` differs.

Reply:
I get a compile error with `++` but not with `++:`.

Reply:
I think you might need parentheses around it like:
```scala
assert(... sameElements (header ++ data.flatten))
```

Reply:
The ++: is a right associative operator, so the compiler can figure out where to put the parentheses automatically.

I'm OK leaving it as is, I suppose if we start using `++:` more we'll start to recognize it more and it won't be so unusual.